include/circt/Dialect/SV/TypeDecl.td

```diff
@@ -222,6 +218,69 @@ def GetModportOp: SVOp<"modport.get", [NoSideEffect]> {
   ];
 }
 
+def AssignInterfaceSignalOp : SVOp<"interface.signal.assign", []> {
+  let summary = "Assign an interfaces signal to some other signal.";
+  let description = [{
+    Use this to continuously assign a signal inside an interface to a
+    value or other signal.
+  
+    ```
+      sv.interface.signal.assign %iface(@handshake_example::@data) 
+        = %zero32 : i32
+    ```
+
+    Would result in the following SystemVerilog:
+    ```
+      assign iface.data = zero32;
+    ```
+  }];
+
+  let arguments = (ins InterfaceType:$iface, FlatSymbolRefAttr:$signalName,
+    AnyType:$rhs);
+
+  let assemblyFormat = [{
+    $iface `(` custom<IfaceTypeAndSignal>(type($iface), $signalName) `)`
+        `=` $rhs attr-dict `:` type($rhs)
+  }];
+
+  let verifier = [{
+    return ::verifySignalExists(iface(), signalNameAttr());
+  }];
+}
+
+def ReadInterfaceSignalOp : SVOp<"interface.signal.read", [NoSideEffect]> {
+  let summary = "Access the data in an interface's signal.";
+  let description = [{
+    This is an expression to access a signal inside of an interface.
+  
+    ```
+      %ifaceData = sv.interface.signal.read %iface
+          (@handshake_example::@data) : i32
+    ```
+
+    Could result in the following SystemVerilog:
+    ```
+      wire [31:0] ifaceData = iface.data;
```

Comment:
Unrelated to this PR, but I didn't know you could do this. Now I'm wondering if we should lower modports like this. It seems like assigning a modport to a wire would play better with the lowering of instances in #302 .

Reply:
`data` is a signal, not a modport. To my knowledge, there is not mechanism for passing a modport around like a value.

Reply:
Oh ok I wasn't sure. If that's the case, then I think the current approach for modports should work.